packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -358,6 +370,13 @@ class CupertinoTextField extends StatefulWidget {
   /// {@macro flutter.widgets.editableText.textAlign}
   final TextAlign textAlign;
 
+  /// Configuration of tool bar options.
+  ///
+  /// If not set, select all and paste will default to be enabled. Copy and cut
+  /// will be disabled if [obscureText] is true. If [readOnly] is true,
+  /// paste and cut will be disabled regardless.
+  final ToolbarOptions toolbarOptions;
```

Comment:
Did you think about asserting or throwing an exception if cut/paste is true and readOnly is true?  I'm not sure which is better, just wondering if you thought about it.

Reply:
It seems weird to me, but asserting the property of a toolbarOptions is not a constant expression. This prevent me from doing the check in constant constructor. I might be able to do the check in build, but do you think it is a good approach?
As of whether we should enforce this, I think we shouldn't because It probably will be very common to switch between readonly and non-readonly. It feel weird to me that if I switch to read only i have to explicit disable the option.
But I do want to enforce it in selectable text though but was thinking whether worth it to make the constructor non-const or checking it outside the constructor.

Reply:
Oh hmm I thought you could add an assert in the initializer list while still being const.  If not then maybe it's not worth it.